### PR TITLE
[0.3.0] Generic Club Endpoints

### PR DESCRIPTION
**Pull request by Galax028**
This PR resolves BACK-66.

**Features**
- GET /clubs/{id}
  Query club by ID: BACK-68
- GET /clubs
  Query clubs: BACK-69
- POST /clubs/{id}/contacts
  Create club contacts: BACK-70

**Fixes**
- Club is now queryable and sortable.
- Google OAuth login now prompts users which account to use (only in debug mode).

### CODE REVIEW

File: mysk-data-api/src/routes/v1/clubs/create_club_contacts.rs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        Err(Error::InternalSeverError(_, _)) => {
+            return Err(Error::EntityNotFound(
+                "Elective subject not found".to_string(),
+                format!("/clubs/{club_id}/contacts"),
+            ));
+        }
+        _ => unreachable!("Club::get_by_id should always return a Detailed variant"),
+    };
+
+    dbg!(&club.staffs);
```
Comment (smartwhatt) [inline mysk-data-api/src/routes/v1/clubs/create_club_contacts.rs]: `dbg!`

File: mysk-data-api/src/routes/v1/clubs/query_clubs.rs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    _: ApiKeyHeader,
+) -> Result<impl Responder> {
+    let pool = &data.db;
+    let fetch_level = request_query.fetch_level.as_ref();
+    let descendant_fetch_level = request_query.descendant_fetch_level.as_ref();
+    let filter = request_query.filter.as_ref();
+    let sort = request_query.sort.as_ref();
+    let pagination = request_query.pagination.as_ref();
+
+    let electives = Club::query(
```
Comment (smartwhatt) [inline mysk-data-api/src/routes/v1/clubs/query_clubs.rs:31]: `electives`?

File: mysk-lib/src/models/club/db.rs [hunk 3]
```diff
@@ -1,36 +1,50 @@
...
 use mysk_lib_macros::traits::db::{BaseQuery, GetById};
 use serde::Deserialize;
-use sqlx::{query, FromRow, PgPool, Postgres, QueryBuilder};
+use sqlx::{query, FromRow, PgPool, Postgres, QueryBuilder, Row as _};
 use uuid::Uuid;
 
 #[derive(BaseQuery, Clone, Debug, Deserialize, FromRow, GetById)]
-#[base_query(query = "SELECT * FROM clubs")]
+#[base_query(
+    query = "
```
Comment (smartwhatt) [inline mysk-lib/src/models/club/db.rs:24]: I suggest creating a view for clubs with this query maybe name it as `detailed_club_view`

**Review by smartwhatt — CHANGES_REQUESTED** [review]

File: mysk-lib/src/models/club/db.rs (hunk 3 above)
Comment (Galax028) [reply mysk-lib/src/models/club/db.rs:24]: At the end of the day that view's query is going to use an `INNER JOIN` anyways, so why don't we use the abstractions here instead?

Comment (smartwhatt) [reply mysk-lib/src/models/club/db.rs:24]: Abstraction on the database level is easier to change and keep track of than on the codebase level

Comment (Galax028) [reply mysk-lib/src/models/club/db.rs:24]: Okay, well here are my counter arguments:
1. Both the teacher and student data models are still using joins in their base queries.
2. Compared to `elective_subject_sessions_with_detailed_view`'s query, club's query is not as complicated that which I personally feel that makes it good enough to be included in the codebase instead.

But, I guess it's your call at the end of the day.

Comment (smartwhatt) [reply mysk-lib/src/models/club/db.rs:24]: Well, the club's supposed to include member_count from compact onwards but that's not required for porting the original codebase, personally all the inherited style tables are supposed to be made a views so data can be query in the same way both using the supabase's SQL editor and using the codebase

**Review by smartwhatt — APPROVED** [review]
LGTM!

**Review by owo93 — APPROVED** [review]
LGTM 🍽️

**Pull request merged**